saddle-registry.md

```diff
@@ -0,0 +1,123 @@
+
+## Master Registry<br>
+
+This contract holds information on all available Saddle Registries.<br> 
+| Network | Address                                                                                                            |
+| :------------ | :------------------------------------------------------------------------------------------------------------------------- |
+| `Ethereum`   | [0xc5ad17b98D7fe73B6dD3b0df5b3040457E68C045](https://etherscan.io/address/0xc5ad17b98D7fe73B6dD3b0df5b3040457E68C045#code) |
+
+<br>
+
+### Master Registry Events
+
+* Retrieve all event of registries that have been added to the Master Registry:
+
+    ``` typescript
+    const hre = require("hardhat");
+    const { ethers } = require("@nomiclabs/hardhat-ethers");
+    import registry_abi from "../master_registry_abi.json";
+
+    async function main() {
+
+        // Initialize contract
+        const registry = await hre.ethers.getContractAt(registry_abi, "0xc5ad17b98D7fe73B6dD3b0df5b3040457E68C045");
+
+        // Get all AddRegistry events
+        const provider = hre.ethers.provider;
+
+        // Optionally you can add a 'topics' field to the filter to only search
+        // for certain events
+        const filter = {
+            address: registry.address,
+            fromBlock: 0,
+            toBlock: 'latest',
+        };
+        const filtered_logs = await provider.getLogs(filter);
+
+        // Setup interface for registry
+        const iface = new hre.ethers.utils.Interface(registry_abi)
+        // Parse each log into a readable event
+        for (let i = 0; i < filtered_logs.length; i++) {
+            const decoded = iface.parseLog(filtered_logs[i]);
+            console.log(decoded);
+        }
+    ```
+<br>
+
+## Pool Registry <br>
+
+This contract holds the funtions to get deployment addresses and related info for all Saddle Pools. It allows for listening to events related to registry actions such as removing/adding pools, as well as any updates to existing pools. <br>
+
+| Network | Address                                                                                                            |
+| :------------ | :------------------------------------------------------------------------------------------------------------------------- |
+| `Ethereum`   | [0xFb4DE84c4375d7c8577327153dE88f58F69EeC81](https://etherscan.io/address/0xFb4DE84c4375d7c8577327153dE88f58F69EeC81#code) |
+
+<br>
+
+### Pool Registry Interactions
+
+The Pool Registry's functions allow you to fetch pool data is, as well as listen for pool events:
+
+* Getting all available PoolData
+
+    ```typescript
+    const hre = require("hardhat");
+    const { ethers } = require("@nomiclabs/hardhat-ethers");
```

Comment:
hardhat shouldn't be necessary in these demo scripts, we can just use ethers directly `import ethers from 'ethers'`